src/components/plots/linePlot.ts

```diff
@@ -104,12 +105,20 @@ export module Plot {
       datasetKeys.forEach((datasetKey: string) => {
         var plotData = this.getAllPlotData(datasetKey);
         plotData.pixelPoints.forEach((pixelPoint: Point, index: number) => {
-          var pixelPointDist = _Util.Methods.distanceSquared(queryPoint, pixelPoint);
-          if (pixelPointDist < closestDistanceSquared) {
-            closestDistanceSquared = pixelPointDist;
-            closestDatum = plotData.data[index];
-            closestPoint = pixelPoint;
-            closestSelection = plotData.selection;
+          var xDist = Math.abs(queryPoint.x - pixelPoint.x);
+          if (xDist <= closestXDistance) {
+            var yDist = Math.abs(queryPoint.y - pixelPoint.y);
+            if (xDist < closestXDistance && yDist < withinValue) {
+              closestXDistance = xDist;
+              closestYDistance = withinValue;
+            }
+
+            if (yDist < closestYDistance) {
```

Comment:
This seems incorrect. If `yDist` is less than the current closest, then the current closest gets overridden even though it might be closer by x.